crates/telio-nurse/src/aggregator.rs

```diff
@@ -0,0 +1,999 @@
+            {
+                Some(ConnectionSegmentData::new_peer(
+                    &entry.get().0,
+                    entry.get().1,
+                    event.timestamp.into(),
+                    event.rx_bytes,
+                    event.tx_bytes,
+                ))
+            }
+            _ => None,
+        };
+
+        if event.peer_state != PeerState::Connected {
+            if let Entry::Occupied(entry) = event_entry {
+                entry.remove();
+            }
+        } else if let Entry::Vacant(entry) = event_entry {
+            entry.insert((event, endpoints));
+        }
+
+        data_guard.peer_segments.extend(new_segment);
+    }
+
+    #[allow(dead_code)]
+    pub async fn change_relay_state(
+        &mut self,
+        event: AnalyticsEvent,
+        reason: RelayConnectionChangeReason,
+    ) {
+        if !self.aggregate_relay_events {
+            return;
+        }
+
+        let mut data_guard = self.data.lock().await;
+
+        let new_segment = data_guard
+            .current_relay_event
+            .as_ref()
+            .filter(|old_event| {
+                old_event.public_key != event.public_key || old_event.peer_state != event.peer_state
+            })
+            .map(|old_event| ConnectionSegmentData::new_relay(old_event, event.timestamp.into()));
+
+        if let Some(segment) = new_segment {
+            data_guard.relay_segments.push(segment);
+        }
+
+        if new_segment.is_some() || data_guard.current_relay_event.is_none() {
+            data_guard.current_relay_event = Some(RelayEvent {
+                public_key: event.public_key,
+                peer_state: event.peer_state,
+                timestamp: event.timestamp.into(),
+                reason,
+            })
+        }
+    }
+
+    #[allow(dead_code)]
+    pub async fn collect_unacknowledged_segments(
+        &self,
+        force_close: bool,
+    ) -> AggregatorCollectedSegments {
+        // Getting wg_interface is first, so it won't await while mutex is locked
+        let wg_interface = self.wg_interface.get_interface().await;
+        let mut data_guard = self.data.lock().await;
+
+        let current_timestamp = Instant::now();
+
+        println!("Relay event: {:?}", data_guard.current_relay_event);
+
+        // Handle long-lasting relay events
+        let mut new_relay_segments = Vec::new();
+        if let Some(event) = data_guard
+            .current_relay_event
+            .as_mut()
+            .filter(|relay_event| {
+                current_timestamp.duration_since(relay_event.timestamp) > DAY_DURATION
+                    || force_close
+            })
+        {
+            new_relay_segments.push(ConnectionSegmentData::new_relay(event, current_timestamp));
+            event.timestamp = current_timestamp;
+        }
+
+        if force_close {
+            data_guard.current_relay_event.take();
```

Comment:
nit: I think `clear` which you use later on is more clear.

Reply:
unfortunately in case of option it seems that `take` is everything we have